libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Teams/Actions/BaseAuthResponseDialog.cs

```diff
@@ -0,0 +1,188 @@
+
+        /// <summary>
+        /// Create a type specific Auth Response using the provided <see cref="CardAction"/>.
+        /// </summary>
+        /// <param name="dc">Current <see cref="DialogContext"/>.</param>
+        /// <param name="cardAction"><see cref="CardAction"/> with a valid Sign In link.</param>
+        /// <returns>An InvokeResponse <see cref="Activity"/> specific to this type of auth dialog.</returns>
+        protected abstract Activity CreateOAuthInvokeResponseActivity(DialogContext dc, CardAction cardAction);
+
+        private static Task<TokenResponse> GetUserTokenAsync(DialogContext dc, string connectionName, CancellationToken cancellationToken)
+        {
+            // When the Bot Service Auth flow completes, the action.State will contain a magic code used for verification.
+            string state = null;
+            if (dc.Context.Activity.Value is JObject valueAsObject)
+            {
+                state = valueAsObject.Value<string>("state");
+            }
+
+            var userTokenClient = dc.Context.TurnState.Get<UserTokenClient>();
+            if (userTokenClient != null)
+            {
+                // CreateTokenResponseFromMessageAsync requires an activity with 'magiccode' in the Text property.
+                var activity = MessageFactory.Text(state);
+                return OAuthHelper.CreateTokenResponseFromMessageAsync(userTokenClient, connectionName, activity, cancellationToken);
+            }
+
+            //TODO: deprecate IExtendedUserTokenProvider
+            if (!(dc.Context.Adapter is IExtendedUserTokenProvider tokenProvider))
+            {
+                throw new InvalidOperationException($"Auth is not supported by the current adapter");
+            }
+            
+            string magicCode = null;
+            if (!string.IsNullOrEmpty(state))
+            {
+                if (int.TryParse(state, out var parsed))
+                {
+                    magicCode = parsed.ToString(CultureInfo.InvariantCulture);
+                }
+            }
+
+            // TODO: SSO and skills token exchange
+            return tokenProvider.GetUserTokenAsync(dc.Context, connectionName, magicCode, cancellationToken: cancellationToken);
+        }
+
+        private async Task<Activity> CreateOAuthInvokeResponseActivityAsync(DialogContext dc, string title, string connectionName, CancellationToken cancellationToken)
+        {
+            var userTokenClient = dc.Context.TurnState.Get<UserTokenClient>();
+            if (userTokenClient != null)
+            {
+                var signInResource = await userTokenClient.GetSignInResourceAsync(connectionName, dc.Context.Activity, null, cancellationToken).ConfigureAwait(false);
+                return CreateOAuthInvokeResponseActivity(dc, title, signInResource.SignInLink);
+            }
+
+            //TODO: deprecate IExtendedUserTokenProvider
+            if (!(dc.Context.Adapter is IExtendedUserTokenProvider tokenProvider))
```

Comment:
Seems like you should actually deprecate it or at least put the comment there.

Reply:
Removed the comment.... at some point, this will be deprecated throughout the sdk